src/LevelAgreementLevel.php

```diff
@@ -414,10 +394,148 @@ public static function displayTabContentForItem(CommonGLPI $item, $tabnum = 1, $
      * the "work_in_day" parameter set to true ?
      *
      * @return bool
+     * @used-by LevelAgreement::computeExecutionDate()
      */
     public function shouldUseWorkInDayMode(): bool
     {
         // No definition time here so we must guess the unit from the raw seconds value
         return abs($this->fields['execution_time']) >= DAY_TIMESTAMP;
     }
+
+    /**
+     * Show the Level Agreement rule form
+     *
+     * {@inheritdoc}
+     **/
+    public function showForm($ID, array $options = [])
+    {
+        /** @var class-string<LevelAgreement> $parent_class */
+        $parent_class = static::$parentclass;
+        $canedit = $this->can($parent_class::$rightname, UPDATE);
+        if (isset($options['la'])) {
+            $la = $options['la'];
+        } else {
+            $la = new $parent_class();
+            $la->getFromDB($this->fields[$parent_class::getForeignKeyField()]);
+        }
+
+        TemplateRenderer::getInstance()->display('pages/setup/levelagreement_level.html.twig', [
+            'item' => $this,
+            'no_header' => $options['no_header'] ?? false,
+            'parent_class' => $parent_class,
+            'la' => $la,
+            'operators' => $this->getRulesMatch(is_string($this->restrict_matching) ? $this->restrict_matching : null),
+            'params' => $options + [
+                'canedit' => $canedit,
+            ],
+        ]);
+    }
+
+    /**
+     * @param LevelAgreement $la The Level Agreement object (SLA or OLA)
+     * @return void
+     */
+    public function showForLA(LevelAgreement $la): void
```

Comment:
It is used only by subclasses, and I do not see any reason to permit this method to be overriden. We can make it final and protected.

```suggestion
    final protected function showForLA(LevelAgreement $la): void
```